src/settings/ActionProfiles/ActionProfilesForm.js

```diff
@@ -22,20 +38,96 @@ import {
   validateRequiredField,
 } from '../../utils';
 import { LAYER_TYPES } from '../../utils/constants';
+import {
+  FolioRecordTypeSelect,
+  ReactToSelect,
+  ACTION_TYPES,
+  ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES,
+} from '../../components';
 
 const formName = 'actionProfilesForm';
 
 export const ActionProfilesFormComponent = ({
+  intl: { formatMessage },
   pristine,
   submitting,
   initialValues,
   handleSubmit,
   location: { search },
   associatedJobProfilesAmount,
   onCancel,
+  action,
+  folioRecord,
 }) => {
   const [isConfirmEditModalOpen, setConfirmModalOpen] = useState(false);
 
+  const getFilteredActions = () => {
+    switch (folioRecord) {
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.ORDER.type: {
+        return pick(ACTION_TYPES, ACTION_TYPES.CREATE.type);
+      }
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.INVOICE.type: {
+        return pick(ACTION_TYPES, [
+          ACTION_TYPES.CREATE.type,
+          ACTION_TYPES.COMBINE.type,
+        ]);
+      }
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.INSTANCE.type:
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.HOLDINGS.type:
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.ITEM.type: {
+        return pick(ACTION_TYPES, [
+          ACTION_TYPES.CREATE.type,
+          ACTION_TYPES.COMBINE.type,
+          ACTION_TYPES.REPLACE.type,
+        ]);
+      }
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.MARC_BIBLIOGRAPHIC.type:
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.MARC_HOLDINGS.type:
+      case ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.MARC_AUTHORITY.type:
+      default: {
+        return ACTION_TYPES;
+      }
+    }
+  };
+
+  const getActionsDataOptions = () => Object.entries(getFilteredActions())
+    .map(([recordType, { captionId }]) => ({
+      value: recordType,
+      label: formatMessage({ id: captionId }),
+    }));
+
+  const getFilteredFolioRecordTypes = () => {
+    switch (action) {
+      case ACTION_TYPES.COMBINE.type: {
+        return omit(ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES, ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.ORDER.type);
+      }
+      case ACTION_TYPES.MODIFY.type: {
+        return pick(ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES, [
+          ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.MARC_AUTHORITY.type,
+          ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.MARC_BIBLIOGRAPHIC.type,
+          ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.MARC_HOLDINGS.type,
+        ]);
+      }
+      case ACTION_TYPES.REPLACE.type: {
+        return omit(ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES, [
+          ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.ORDER.type,
+          ACTION_PROFILES_FORM_FOLIO_RECORD_TYPES.INVOICE.type,
+        ]);
+      }
+      case ACTION_TYPES.CREATE.type:
```

Comment:
Not a big deal, but this line is excess really since it is handled by the default case.

Reply:
I agree but I did it intentionally for readability purposes. Here and in the function above.